### PR TITLE
Add xgboost example using Bayesian optimization

### PR DESCRIPTION
**Pull request by richardsliu**
This example uses the image built from https://github.com/kubeflow/examples/tree/master/xgboost_ames_housing.

The hyperparameters being tuned are:
- Number of boost trees
- learning rate

We are trying to minimize the mean_absolute_error.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/320)

### CODE REVIEW

**Comment by johnugeorge** [comment]
Is it using default metric collector?

**Comment by richardsliu** [comment]
@johnugeorge Yes, the default metric collector works here (assuming that the training worker outputs logs in the expected format).

**Comment by richardsliu** [comment]
@hougangliu The example docker image is being fixed in this PR: https://github.com/kubeflow/examples/pull/476. I added comments clarifying the prerequisites.
